src/audio/dai-zephyr.c

```diff
@@ -438,182 +438,136 @@ static int dai_verify_params(struct comp_dev *dev, struct sof_ipc_stream_params
 	return 0;
 }
 
-/* set component audio SSP and DMA configuration */
-static int dai_playback_params(struct comp_dev *dev, uint32_t period_bytes,
-			       uint32_t period_count)
+static int dai_set_dma_buffer(struct comp_dev *dev, struct sof_ipc_stream_params *hw_params,
+			      uint32_t *pb, uint32_t *pc)
```

Comment:
if you just move this function to right above `dai_params()` the diffstat becomes some 40 lines smaller and easier to review